sympy/calculus/util.py

```diff
@@ -1629,3 +1532,133 @@ def union(self, other):
 
 # setting an alias for AccumulationBounds
 AccumBounds = AccumulationBounds
+
+# Hooks
+def Add_hook(*args, **options):
+    evaluate = options.get('evaluate', global_parameters.evaluate)
+    if not evaluate:
+        return None
+
+    extras, accbnds, others = [], [], []
+    for a in args:
+        if isinstance(a, MatrixExpr):
+            extras.append(a)
```

Comment:
Should this instead be `if not a.is_extended_real`? As written, this looks like exactly the special case you wanted to avoid.